docs/gen-ai/llm-metrics.md

```diff
@@ -0,0 +1,114 @@
+<!--- Hugo front matter used to generate the website version of this page:
+linkTitle: LLM metrics
+--->
+
+# Semantic Conventions for LLM Metrics
+
+**Status**: [Experimental][DocumentStatus]
+
+The conventions described in this section are specific to LLM clients.
+
+**Disclaimer:** These are initial LLM client metric instruments
+and attributes but more may be added in the future.
+
+<!-- Re-generate TOC with `markdown-toc --no-first-h1 -i` -->
+
+<!-- toc -->
+
+- [LLM Requests](#llm-requests)
+  - [Metric: `gen_ai.usage.tokens`](#metric-gen_aiusagetokens)
+  - [Metrics: `gen_ai.request.duration`](#metrics-gen_airequestduration)
+
+<!-- tocstop -->
+
+## LLM Requests
+
+The following metric instruments describe LLM request operations.
+
+### Metric: `gen_ai.usage.tokens`
+
+This metric is [required][MetricRequired].
+
+<!-- semconv metric.gen_ai.usage.tokens(metric_table) -->
+| Name     | Instrument Type | Unit (UCUM) | Description    | Stability |
+| -------- | --------------- | ----------- | -------------- | --------- |
+| `gen_ai.usage.tokens` | Histogram | `tokens` | LLM usage attributes | ![Experimental](https://img.shields.io/badge/-experimental-blue) |
```

Comment:
I also wonder if this should be a counter instead of a histogram, with backends needing to sum the values to show current usage. The downside is it's harder to see a distribution like you would with a histogram.

Reply:
we have a dimension that separates completion vs prompt - `gen_ai.usage.token_type` and they sum up to the total. 

This matches other metrics in otel where `*.usage` follows this patten (see https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/metrics.md#instrument-naming).

So from number of instruments/naming, I'd suggest to keep one instrument with the dimension, but call it `gen_ai.tokens.usage` to match the otel pattern.

Reply:
Counter vs histogram is an interesting one. 

Histogram allows to capture the distribution (e.g. P95 of prompts and completions) and find outliers (combined with exemplars, I can find traces and ultimately examples of cases when the completion was extremely small or big). This is super-useful.

Histograms also have a counter time series that represents total count, so it does allow everything that counters allow.

Reply:
@TaoChenOSU @cartermp This is a design point. I like the combined single metric and then have the type of token as prompt, completion as an attribute. This also supports having "input" as a type which is how OpenAI refers to embedding tokens. Are there advantages to having these split into separate instruments?

@lmolkova good point on the name. I was following our attribute naming which was gen_ai.usage.*_tokens, but following the metric convention is better.

Reply:
My concern was mainly ease of use. Having individual metrics could make things more obvious for users. Nonetheless, I wasn't aware of the established pattern. I think we should keep the consistency.

Reply:
Thanks all! Agreed with the overall design chosen here. Thanks for elaborating.